Batteries/Data/Char/Basic.lean

```diff
@@ -22,3 +26,102 @@ theorem toNat_ofNat (n : Nat) : toNat (ofNat n) = if n.isValidChar then n else 0
   split
   · simp [ofNat, *]
   · simp [ofNat, toNat, *]
+
+/-- Returns `true` if `p` returns true for every `Char`. -/
+protected def all (p : Char → Bool) : Bool :=
+  -- 0x10FFFF (1114111) is the maximal valid code point.
```

Comment:
Could assertion such as this come with a link to a source?

Reply:
Presumably the source is just `UInt32.isValidChar`? In practice the unicode spec isn't relevant, only that this enumerates every `Char`.